srtp/srtp.c

```diff
@@ -4841,8 +4841,125 @@ srtp_err_status_t srtp_get_stream_roc(srtp_t session,
 
 #ifndef SRTP_NO_STREAM_LIST
 
+#ifdef ENABLE_STREAM_INDEX
+
+#define INITIAL_STREAM_INDEX_SIZE 2
+
+typedef struct stream_index_entry {
+    uint32_t ssrc;
+    srtp_stream_t stream;
+} stream_index_entry;
+
+typedef struct srtp_stream_index_ {
+    stream_index_entry *entries;
+    size_t size;
+    size_t available;
+} srtp_stream_index_;
+
+srtp_err_status_t srtp_stream_index_alloc(srtp_stream_index *stream_index_ptr)
+{
+    srtp_stream_index stream_index =
+        srtp_crypto_alloc(sizeof(srtp_stream_index_));
+    if (stream_index == NULL) {
+        return srtp_err_status_alloc_fail;
+    }
+
+    stream_index->entries = srtp_crypto_alloc(sizeof(stream_index_entry) *
+                                              INITIAL_STREAM_INDEX_SIZE);
+    if (stream_index->entries == NULL) {
+        return srtp_err_status_alloc_fail;
```

Comment:
If you exit here you'd leak `stream_index`, wouldn't you?

Reply:
correct :+1:

Reply:
I'll fix it.

Reply:
done, good catch @saghul ❤️